learner_pathway_progress/utilities.py

```diff
@@ -0,0 +1,188 @@
+            'course_id',
+            'enterprise_customer_user__user_id',
+        )
+        log.info(
+            f"{PATHWAY_LOGS_IDENTIFIER}: Enterprise Course Enrollment with offset:"
+            f" {offset} and offset chunk size {offset + CHUNK_SIZE}"
+        )
+        for pathway in pathways:
+            pathway_courses = pathway['courses']
+            for enrollment in ent_course_enrollments:
+                user = User.objects.filter(id=enrollment.enterprise_customer_user.user_id).first()
+                pathway_progress_exist = LearnerPathwayProgress.objects.filter(
+                    user=user,
+                    learner_pathway_uuid=pathway['pathway-uuid'],
+                ).exists()
+                if not pathway_progress_exist and enrollment.course_id in pathway_courses:
+
+                    LearnerPathwayMembership.objects.get_or_create(
+                        user=user,
+                        learner_pathway_uuid=pathway['pathway-uuid']
+                    )
+                    pathway_progress = LearnerPathwayProgress.objects.create(
+                        user=user,
+                        learner_pathway_uuid=pathway['pathway-uuid'],
+                    )
+                    pathway_progress.learner_pathway_progress = json.dumps(pathway['snapshot'])
+                    pathway_progress.save()
+                    pathway_progress.update_pathway_progress()
+
+
+def get_pathway_course_and_course_run_keys(step_courses, step_programs, courses):
+    """Return all the course run keys linked with pathway step."""
+    for course in step_courses:
+        course_runs = course.get('course_runs') or []
+        for course_run in course_runs:
+            courses.append(course_run['key'])
+    for program in step_programs:
+        program_courses = program.get('courses') or []
+        for p_course in program_courses:
+            p_course_runs = p_course.get('course_runs') or []
+            for p_course_run in p_course_runs:
+                courses.append(p_course_run['key'])
+
+
+def update_all_pathways_progress():
+    """Update all pathways progress."""
+    # pylint: disable=unused-variable
+    user, catalog_integration = check_catalog_integration_and_get_user(
+        error_message_field='Call to fetch pathways from pathway progress management command'
+    )
+    if user:
+        learner_pathways = get_all_learner_pathways(user.id)
+        pathway_data = {"data": []}
+        pathway_snapshots = learner_pathways.get('results') or []
+        for pathway_snapshot in pathway_snapshots:
+            pathway_details = {}
+            courses = []
+            pathway_steps = pathway_snapshot.get('steps') or []
+            for step in pathway_steps:
+                step_courses = step.get('courses') or []
+                step_programs = step.get('programs') or []
+
+                get_pathway_course_and_course_run_keys(step_courses, step_programs, courses)
```

Comment:
Shouldn't the last argument name be 'course_run_keys' as that is what is returned